### ISSUE TITLE
Numerous compiler errors occur at startup unless I set skipLibCheck to true

### ISSUE THREAD

**Issue by cdoremus** [wmonk/create-react-app-typescript#53]
If you are reporting a bug, please fill in below. Otherwise feel free to remove this template entirely.

### Can you reproduce the problem with latest npm? 
**Yes**
Many errors, especially related to "missing modules", are due to npm bugs.

If you're using Windows, [follow these instructions to update npm](https://github.com/npm/npm/wiki/Troubleshooting#upgrading-on-windows).

If you're using OS X or Linux, run this to update npm:

```
npm install -g npm@latest

cd your_project_directory
rm -rf node_modules
npm install
```

Then try to reproduce the issue again.

Can you still reproduce it? **Yes**
### Description

What are you reporting?
**I get numerous compiler errors at application startup (yarn start) when I use create-react-app with the react-scripts-ts script (see attached file showing errors). However, these errors are resolved if I set skipLibCheck to true in tsconfig.json.**

[CompileErrorsWhenNotUsing_skipLibCheck-compilerOption.txt](https://github.com/wmonk/create-react-app-typescript/files/999916/CompileErrorsWhenNotUsing_skipLibCheck-compilerOption.txt)

### Expected behavior

Tell us what you think should happen. 
**No errors should occur upon initial startup and I should not need to add skipLibCheck=true to tsconfig.json**

### Actual behavior

Tell us what actually happens. **This is described above**

### Environment

Run these commands in the project folder and fill in their results:

1. `npm ls react-scripts` (if you haven’t ejected): 
```
react-ts-test@0.1.0 /home/craig/git/react-ts-test
└── (empty)
```
2. `node -v`: ```v7.6.0```

3. `npm -v`: ```4.5.0```

Then, specify:

1. Operating system: ```Ubuntu 16.04 LTS```
2. Browser and version: ```Chrome 58.0.3029.110 (64-bit)```

### Reproducible Demo

Please take the time to create a new app that reproduces the issue.

Alternatively, you could copy your app that experiences the problem and start removing things until you’re left with the minimal reproducible demo.

(Accidentally, you might get to the root of your problem during that process.)

Push to GitHub and paste the link here. **The repo is attached in a zip file (minus node_modules)**
[react-ts-test.zip](https://github.com/wmonk/create-react-app-typescript/files/999918/react-ts-test.zip)

By doing this, you're helping the Create
 React App contributors a big time!
Demonstrable issues gets fixed faster.

**Comment by AJamesPhillips** [comment]
Hi @cdoremus can you post some of the errors your seeing please?  What command are you issuing and from which directory?  What version of typescript are you running?  Thanks.

**Comment by cdoremus** [comment]
There is a file attached to the issue above that shows the compile errors. As instructed in the README, I ran the command `create-react-app react-ts-test --scripts-version=react-scripts-ts` followed by `cd my-react-ts-test` and then `yarn start`. My TypeScript version is 2.3.2 and I installed `react-scripts-ts` 1.4.0 globally. My version of `create-react-app` is 1.3.1.

**Comment by DanielRosenwasser** [comment]
I tried `yarn start` on the sample that you gave and could not reproduce the errors that you're encountering at all. Even a plain compile with `tsc` hasn't given me these issues.

My thinking is that for some reason there are potentially two versions of `@types/react` being found on your system somehow that are conflicting. A reasonable way to check this might be to dive into `/home/craig/node_modules/@types/react/index.d.ts`, going to one of the declarations with an error (e.g. `textPath`) and finding all references on there.

Can you let me know what you end up finding?

**Comment by wmonk** [comment]
Closing as this has not been updated in a while.

**Comment by wmonk** [comment]
@cdoremus just re-read your comment - did you manage to get this working. I see that you installed `react-script-ts` globally, which you definitely don't need to do. `react-scripts-ts` is installed locally to your project, which as @DanielRosenwasser suggests might mean you have duplicated versions of `@types/react`.
